src/DataStructures/DataBox/DataBox.hpp

```diff
@@ -297,8 +297,30 @@ template <typename TagList, typename Tag>
 using has_subitems = tmpl::not_<
     std::is_same<typename Subitems<TagList, Tag>::type, tmpl::list<>>>;
 
+template <typename Caller, typename Argument, typename IsOk>
+struct report_missing_compute_item_argument {
+  static_assert(cpp17::is_same_v<Caller, void>,
+                "A compute item's argument cannot be found in the DataBox.  "
```

Comment:
Wouldn't this also be triggered if there are two of the same base tags in the DataBox? If so, then the message should mention that